src/IoC/InstancePool.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace IoC;
+
+public sealed class InstancePool
+{
+    //This is where the instances will be stored.
+    private static Dictionary<string, object>? _implementationStore;
+    //Make the contructor unaccessible.
+    InstancePool()
+    {
+        _implementationStore = new Dictionary<string, object>();
+    }
+
+
+    //This is the only way to access to the methods of the class.
+    private static InstancePool? _instance = null;
+    public static InstancePool Instance
+    {
+        get
+        {
+            if (_instance is null)
+            {
+                _instance = new InstancePool();
+            }
+            return _instance;
+        }
+    }
+
+    //This method register the class.
+    public void Register<TClass>() where TClass : class
```

Comment:
Do you think this is a good idea? I feel when you instance a class you may need special and custom logic, I feel it is better to:

1. Allow the user return the instance of the class by using delegates
2. Build a factory pattern that will allow the user to implement their creation factory (I dont like this one a lot because it's too complex and too big for something little)
3. Create interfaces and only allow to use implementations of this interface to register the classes.

I just say that this implementation is tol generic that maybe useless in a close future. Also how would you set dependencies for a specific class?

These are just questions feel free to explain what is happening here.

Reply:
WIP. Expect to have it working by tomorrow @reynoldmorel.